src/commands/config/create.ts

```diff
@@ -0,0 +1,239 @@
+      success(`Token registry deployed, address: ${highlight(contractAddress)}`);
+      return contractAddress;
+    };
+
+    const createTemporaryDnsWithDocumentStore = async (documentStoreAddress: string): Promise<string> => {
+      const documentStoreTemporaryDnsParams = {
+        networkId: 3,
+        address: documentStoreAddress,
+        sandboxEndpoint: sandboxEndpointUrl,
+      };
+      info(`Creating temporary DNS for verifiable documents with DNS-TXT`);
+      return (await createTemporaryDns(documentStoreTemporaryDnsParams)) || "";
+    };
+
+    const createTemporaryDnsWithDid = async (): Promise<string> => {
+      const didTemporaryDnsParams = {
+        networkId: 3,
+        publicKey: `did:ethr:0x${walletObject.address}#controller`,
+        sandboxEndpoint: sandboxEndpointUrl,
+      };
+      info(`Creating Creating temporary DNS for verifiable documents with DNS-DID`);
+      return (await createTemporaryDns(didTemporaryDnsParams)) || "";
+    };
+
+    const createTemporaryDnsWithTokenRegistry = async (tokenRegistryAddress: string): Promise<string> => {
+      const tokenRegistryTemporaryDnsParams = {
+        networkId: 3,
+        address: tokenRegistryAddress,
+        sandboxEndpoint: sandboxEndpointUrl,
+      };
+      info(`Creating temporary DNS for transferable records`);
+      return (await createTemporaryDns(tokenRegistryTemporaryDnsParams)) || "";
+    };
+
+    // loop through the form template to check the type of forms
+    const typesOfForms = formsInTemplate.map((form) => {
+      const identityProofTypes = form.defaults.issuers.map((issuer: Issuer) => issuer.identityProof?.type);
+      return {
+        type: form.type,
+        identityProofTypes: identityProofTypes,
+      };
+    });
+
+    // generate doc store or token registry based on the form type in the form template
+    const documentStoreAddress =
+      typesOfForms.filter(
+        (item) =>
+          item.type === "VERIFIABLE_DOCUMENT" && (<any>Object).values(item.identityProofTypes).includes("DNS-TXT")
+      ).length > 0
+        ? await createDocumentStore()
+        : "";
+    const verifiableDocumentDnsTxtName = documentStoreAddress
+      ? await createTemporaryDnsWithDocumentStore(documentStoreAddress)
+      : "";
+    const verifiableDocumentDnsDidName =
+      typesOfForms.filter(
+        (item) =>
+          item.type === "VERIFIABLE_DOCUMENT" && (<any>Object).values(item.identityProofTypes).includes("DNS-DID")
+      ).length > 0
+        ? await createTemporaryDnsWithDid()
+        : "";
+
+    const tokenRegistryAddress =
+      typesOfForms.filter(
+        (item) =>
+          item.type === "TRANSFERABLE_RECORD" && (<any>Object).values(item.identityProofTypes).includes("DNS-TXT")
+      ).length > 0
+        ? await createTokenRegistry()
+        : "";
+    const tokenRegistryDnsName = tokenRegistryAddress
+      ? await createTemporaryDnsWithTokenRegistry(tokenRegistryAddress)
```

Comment:
isn't it equal to createTemporaryDnsWithDocumentStore ?

Reply:
ok will refactor the similar functions